h/static/scripts/controllers/share-widget-controller.js

```diff
@@ -0,0 +1,211 @@
+           *   with the correct information per card.
+           */
+          _renderWidgetTemplate: function(config) {
+
+            // copy to clipboard input
+            this._widget.querySelector(CLIPBOARD_INPUT_SELECTOR).value = config.url;
+
+            // social links
+            Array.from(this._widget.querySelectorAll(TARGET_HREF_SELECTOR)).forEach((target) => {
+              target.href = target.getAttribute(TARGET_HREF_ATTR).replace('{href}', encodeURI(config.url));
+            });
+
+            // scope access dialog
+            const privateMessage = this._widget.querySelector(PRIVATE_MSG_SELECTOR);
+            const groupMessage = this._widget.querySelector(GROUP_MSG_SELECTOR);
+
+            privateMessage.style.display = 'none';
+            groupMessage.style.display = 'none';
+
+            if (config.private) {
+              privateMessage.style.display = 'block';
+            } else if (config.group) {
+              groupMessage.style.display = 'block';
+            }
+          },
+
+
+          /**
+           * Given a node, update the template, repostion the widget properly,
+           *  and make it visible.
+           *
+           * @param  {HTMLNode} node The trigger node that we will place the widget
+           *   next to.
+           * @param  {ConfigOptions} config Passed through to rendering/interpolation
+           */
+          showForNode: function(node, config) {
+
+            if (!node || !config) {
+              return;
+            }
+
+            this._renderWidgetTemplate(config);
+
+            // offsets affecting height need to be updated after variable replacement
+            const widgetOffsets =  _getOffset(this._widget);
+            const nodeOffset = _getOffset(node);
+
+            // console.log(this._widget, nodeOffset.top, widgetOffsets.height, ARROW_PADDING_BOTTOM);
+
+            this._widget.style.top = (nodeOffset.top - widgetOffsets.height - ARROW_PADDING_BOTTOM) + 'px';
+            this._widget.style.left = ((ARROW_PADDING_RIGHT + nodeOffset.left + (nodeOffset.width / 2)) - widgetOffsets.width) + 'px';
+
+            this._container.style.visibility = 'visible';
+
+            this._widgetVisible = true;
+          },
+
+          hide: function() {
+            this._container.style.visibility = 'hidden';
+            this._widgetVisible = false;
+          },
+
+
+          init: function(containerElement) {
+
+            this._container = containerElement;
+            this._widget = this._container.querySelector(WIDGET_SELECTOR);
+
+            // on initialize we need to reset container visbility
+            this.hide();
+
+
+            window.document.body.addEventListener('click', (event) => {
+              const target = event.target;
+
+              // do nothing if we are clicking inside of the widget
+              if (containerElement.contains(target) || containerElement === target) {
```

Comment:
This surprised me, but [Element#contains](https://dom.spec.whatwg.org/#dom-node-contains) is _inclusive_ so `someElement.contains(someElement)` returns true and you don't need the second branch of the `if` statement here.